### PR TITLE
Use reproducibility flags

### LINKED ISSUE #311: Use reproducibility flags for Gadi
Document the reproducibility flags that were used in testing https://github.com/penguian/1deg_jra55_ryf migration from NCI Raijin to Gadi.

Using these flags with the Intel 2019.3.199 compilers produced identical output on Raijin and Gadi. See "repro group 4" in [the report summary](https://docs.google.com/document/d/1iFIH_kHkoDJjw4WHOfTqC0ylBRtR4g9O-jX3WSUYn9A/edit?usp=sharing) for results.

### PR DESCRIPTION
**Pull request by penguian**
Use reproducibility flags to allow Broadwell (Raijin) and Cascade Lake (Gadi) to give bit-identical results. Tested on Gadi.

Closes #311

### CODE REVIEW

File: bin/mkmf.template.nci [hunk 1]
```diff
@@ -38,14 +38,15 @@ endif
 
 FPPFLAGS := -fpp -Wp,-w $(INCLUDE)
 FFLAGS := -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -align all
-FFLAGS_OPT = -O2 -debug minimal -xHost
-FFLAGS_DEBUG = -g -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv
-FFLAGS_REPRO = -fp-model precise
-FFLAGS_VERBOSE = -v -V -what
+FFLAGS_OPT := -g3 -O2 -axCORE-AVX2 -debug all -check none -qopt-report=5 -qopt-report-annotate
+FFLAGS_DEBUG := -g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv
+FFLAGS_REPRO := -fp-model consistent -fp-model source -align all
```
Comment (aekiss) [inline bin/mkmf.template.nci]: why the change from `-fp-model precise` to `-fp-model consistent`? `consistent` is probably slower. Is it needed for reproducible results between Broadwell and Cascade Lake?

Comment (penguian) [reply bin/mkmf.template.nci]: Well spotted. I will change to precise and re-test.

Comment (penguian) [reply bin/mkmf.template.nci]: Addressed by commit 36ed508

**Comment by aekiss** [comment]
@penguian I'm guessing we can close this one too?
